mlos_bench/mlos_bench/services/remote/azure/azure_saas.py

```diff
@@ -63,22 +63,20 @@ def __init__(self,
             "subscription",
             "resourceGroup",
             "provider",
+            "supportsBatchUpdate",
+            "isFlex",
+            "apiVersion"
         })
 
-        (self._is_batch, is_flex, api_version) = {
-            "Microsoft.DBforMySQL": (True, True, "2022-01-01"),
-            "Microsoft.DBforMariaDB": (False, False, "2018-06-01"),
-            "Microsoft.DBforPostgreSQL": (False, True, "2022-12-01"),
-        }[self.config["provider"]]
```

Comment:
I think we can keep this as a convenience fallback for when those values aren't provided in the config BTW.

Reply:
oh, you're right. Let me bring it back so we have both ways to configure things